generators/Generators.fs

```diff
@@ -267,6 +267,96 @@ type RailFenceCipher() =
 type Raindrops() =
     inherit Exercise()
 
+type RobotSimulator() =
+    inherit Exercise()
+
+    member this.ResultIdentifierName = "actual"
+
+    override this.PropertiesWithIdentifier canonicalDataCase = ["robot"; "property"; "expected"]
+
+    member private this.RenderDirection (value: JToken) = 
+        sprintf "%s" (value.ToObject<string>() |> String.upperCaseFirst)  
+
+    member private this.RenderCoords (coords: JToken) = 
+        (coords.["x"].ToObject<int>(), coords.["y"].ToObject<int>()) 
+        |> formatValue
+
+    member private this.DefineRobot (direction: JToken) (coords: JToken) = 
+        sprintf "createRobot %s %s" (this.RenderDirection direction) (this.RenderCoords coords)
+
+    member private this.GetRobotProperties (value : JToken) = 
+        value.SelectToken("direction"), value.SelectToken("position")
+
+    override this.RenderArrange canonicalDataCase =
+        // one identifier may be empty if we only checking created object
+        // we need to filter out this empty line
+        base.RenderArrange canonicalDataCase
+        |> List.choose (
+            function
+            | "" -> None
+            | v -> Some v
+        )
+
+    override this.RenderValueWithoutIdentifier (canonicalDataCase, key, value) = 
+        match key with
+        | "robot" ->
+            let input = value :?> JToken
+            this.DefineRobot ("direction" |> input.SelectToken) ("position" |> input.SelectToken)
+        | "expected" -> 
+            // here we may need to render full robot object
+            // or just coordinate/direction values
+            let input = (value :?> JToken)
+            let direction, position = this.GetRobotProperties input
+            match isNull(direction), isNull(position) with
```

Comment:
Do you need the parentheses? Or could `isNull(direction)` be `isNull direction`? The same applies to the second `isNull` call.